src/ILogger/ApplicationInsightsLogger.cs

```diff
@@ -0,0 +1,193 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ApplicationInsightsLogger.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2013
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Extensions.Logging.ApplicationInsights
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Text;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.DataContracts;
+
+    /// <summary>
+    /// Application insights logger implementation for <see cref="ILogger"/>.
+    /// </summary>
+    /// <seealso cref="ILogger" />
+    public class ApplicationInsightsLogger : ILogger
+    {
+        private readonly string categoryName;
+        private readonly TelemetryClient telemetryClient;
+        private readonly ApplicationInsightsLoggerOptions applicationInsightsLoggerOptions;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="ApplicationInsightsLogger"/>.
+        /// </summary>
+        public ApplicationInsightsLogger(
+            string categoryName,
+            TelemetryClient telemetryClient,
+            ApplicationInsightsLoggerOptions applicationInsightsLoggerOptions)
+        {
+            this.categoryName = categoryName;
+            this.telemetryClient = telemetryClient;
+            this.applicationInsightsLoggerOptions = applicationInsightsLoggerOptions ?? throw new ArgumentNullException(nameof(applicationInsightsLoggerOptions));
+        }
+
+        /// <summary>
+        /// Gets or sets the external scope provider.
+        /// </summary>
+        internal IExternalScopeProvider ExternalScopeProvider { get; set; }
+
+        /// <summary>
+        /// Begins a logical operation scope.
+        /// </summary>
+        /// <typeparam name="TState">Current state.</typeparam>
+        /// <param name="state">The identifier for the scope.</param>
+        /// <returns>
+        /// An IDisposable that ends the logical operation scope on dispose.
+        /// </returns>
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            return this.ExternalScopeProvider != null ? this.ExternalScopeProvider.Push(state) : NullScope.Instance;
```

Comment:
We need to do this only if (this.ExternalScopeProvider != null)

Reply:
Method returns an IDisposable regardless so I need to return a dummy object.